.github/workflows/code_quality.yml

```diff
@@ -144,3 +144,30 @@ jobs:
 
       - name: Run linter
         run: python ./bin/lint_dependencies.py
+
+  type-checking:
+    name: Type checking (Pyright)
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Install additional dependencies
+        run: |
+          pip install uv
+          uv pip install \
+            --system \
+            -r requirements/ci.txt \
+            django-stubs \
+            django-stubs-ext \
+            djangorestframework-stubs \
+            types-lxml \
+            types-pytz \
+            types-pyyaml \
+            types-requests
```

Comment:
Should we have an extra `.in` file for these instead?

Reply:
yeah, I'll set that up, will make things easier for local checks too